tests/test_helpers.py

```diff
@@ -190,5 +193,41 @@ def test_extract_product_name(self):
 
         self.assertEqual(short_name_want, short_product_name)
 
+
+    # Filenames to test markdown syntax highlight with.
+    test_markdown_filenames = [
+        [
+            "tests/markdown_syntax_highlight.md",
+            "tests/markdown_syntax_highlight_want.md"
+        ],
+        [
+            "tests/markdown_no_highlight.md",
+            "tests/markdown_no_highlight_want.md"
+        ],
+        [
+            "tests/markdown_mixed_highlight.md",
+            "tests/markdown_mixed_highlight_want.md"
+        ],
+    ]
+    @parameterized.expand(test_markdown_filenames)
+    def test_highlight_md_codeblocks(self, base_filename, want_filename):
+        # Test to ensure codeblocks in markdown files are correctly highlighted.
+
+        # Copy the base file we'll need to test.
+        test_file = tempfile.NamedTemporaryFile(mode='r+', delete=False)
```

Comment:
Use as a context manager (`with ...`) to avoid having to manually close? Why not auto-delete? Debugging if it goes wrong?

Reply:
The file needed to be kept within the function being tested, so I didn't think it'd work to use it as a context manager. Seems to work, updated to use.